File: src/secrets.ts
const FORBIDDEN = ["private_key", "privkey", "mnemonic", "seed", "wif", "secret_key"];

function norm(name: string): string {
  return name.toLowerCase().replace(/[^a-z0-9_]/g, "");
}

export function forbiddenFields(obj: unknown, prefix = ""): string[] {
  const hits: string[] = [];
  if (obj && typeof obj === "object" && !Array.isArray(obj)) {
    for (const [k, v] of Object.entries(obj as Record<string, unknown>)) {
      const path = prefix ? `${prefix}.${k}` : k;
      if (FORBIDDEN.some((n) => norm(k).includes(n))) {
        hits.push(path);
      }
      hits.push(...forbiddenFields(v, path));
    }
  }
  return hits;
}

File: x402stall/secrets.py
from __future__ import annotations

FORBIDDEN_SUBSTR = ("private_key", "privkey", "mnemonic", "seed", "wif", "secret_key")


def forbidden_fields(obj) -> list[str]:
    hits = []

    def walk(o, prefix=""):
        if isinstance(o, dict):
            for k, v in o.items():
                path = f"{prefix}.{k}" if prefix else str(k)
                n = "".join(ch for ch in str(k).lower() if ch.isalnum() or ch == "_")
                if any(s in n for s in FORBIDDEN_SUBSTR):
                    hits.append(path)
                walk(v, path)

    walk(obj)
    return hits
